vllm/v1/engine/coordinator.py

```diff
@@ -183,11 +183,12 @@ def process_input_socket(self, front_publish_address: str,
                     # engines are paused, so that we can wake the other
                     # engines.
                     engine_to_exclude, wave = msgspec.msgpack.decode(buffer)
-                    if wave < self.current_wave:
-                        # If the wave number is stale, ensure the message is
-                        # handled by all the engines.
-                        engine_to_exclude = None
                     if not self.engines_running:
+                        if wave < self.current_wave:
+                            # If the wave number is stale, ensure the message
+                            # is handled by all the engines.
+                            engine_to_exclude = None
```

Comment:
This is unrelated simplification, this check only applies to the case inside the `if`.